### PR TITLE
Add a page listing platform adaptations

### PR DESCRIPTION
**Pull request by xster**
https://flutter-website-staging-6f2ad.firebaseapp.com/docs/resources/platform-adaptations

I sent you an email to get your opinion on the format of the video clips but the text is ready for review. 

There will be a video for each heading.

### CODE REVIEW

File: src/_data/sidenav.yml [hunk 1]
```diff
@@ -192,6 +192,8 @@
       permalink: /docs/resources/bootstrap-into-dart
     - title: Inside Flutter
       permalink: /docs/resources/inside-flutter
+    - title: Platform Adaptations
```
Comment (sfshaza2) [inline src/_data/sidenav.yml]: We use sentence case for titles and headings.  Platform Adaptations => Platform adaptations.

File: src/docs/resources/platform-adaptations.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
+---
+title: Platform Specific Behaviors and Adaptations
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Platform-specific behaviors and adaptations

File: src/docs/resources/platform-adaptations.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+---
+
+## Adaptation Philosophy
+
+There are generally 3 categories of platform adaptiveness:
+
+1. Things that are behaviors of the OS environment (such as text editing and
+scrolling) and that would be 'wrong' a different behavior took place.
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: ..and that would be 'wrong' IF a different...

File: src/docs/resources/platform-adaptations.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+2. Things that are conventionally implemented in apps using the OEM's SDKs
+(such as using parallel tabs on iOS or showing an [android.app.AlertDialog](https://developer.android.com/reference/android/app/AlertDialog.html)
+on Android).
+3. Correct usage of corresponding platform APIs to export to TalkBack/VoiceOver,
+status bar color theme etc. and to import notch metrics, locale settings etc.
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: Ack. This sentence has TWO "etc."s in it. It shouldn't have any, but definitely not two!  Also, add a comma after "status bar". Currently, this sentence is hard to parse.

File: src/docs/resources/platform-adaptations.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+This article mainly covers the automatic adaptations provided by Flutter
+in case 1 on Android and iOS.
+
+For case 2, Flutter bundles the means to produce the appropriate effects of
+the platform conventions but does not adapt automatically when app design
+choices are needed. For a discussion, see [#8410](https://github.com/flutter/flutter/issues/8410#issuecomment-468034023).
+
+Flutter should transparently work for scenario 3.
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: You've called them categories, cases, and scenarios. Please be consistent.

File: src/docs/resources/platform-adaptations.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+### Navigation Transitions
+
+On **Android**, the default [Navigator.push](https://docs.flutter.io/flutter/widgets/Navigator/push.html)
+transition is modeled after [startActivity()](https://developer.android.com/reference/android/app/Activity.html#startActivity(android.content.Intent))
+which generally has one bottom-up animation variant.
+
+On **iOS**:
+
+* The default [Navigator.push](https://docs.flutter.io/flutter/widgets/Navigator/push.html)
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: As in the other PR, globally change docs.flutter.io to api.flutter.dev.

File: src/docs/resources/platform-adaptations.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+On **Android**, 2 page transition animation styles exist depending on your OS
+version:
+
+* Pre API 28 uses a bottom-up animation that [slides up and fades in](https://docs.flutter.io/flutter/material/FadeUpwardsPageTransitionsBuilder-class.html). 
+* On API 28 and later, the bottom-up animation [slides and clip-reveals up](https://docs.flutter.io/flutter/material/OpenUpwardsPageTransitionsBuilder-class.html).
+
+On **iOS**, Flutter's bundled [CupertinoNavigationBar](https://docs.flutter.io/flutter/cupertino/CupertinoNavigationBar-class.html)
+and [CupertinoSliverNavigationBar](https://docs.flutter.io/flutter/cupertino/CupertinoSliverNavigationBar-class.html)
+will automatically animate each of its subcomponent to its corresponding
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: will automatically animate => automatically animates

File: src/docs/resources/platform-adaptations.md (hunk 6 above)
Comment (sfshaza2) [reply src/docs/resources/platform-adaptations.md]: Actually let me correct myself. We don't use explicit URLs for API docs. Please use {{site.api}}/flutter/widgets/...

(This macro is relatively new, but it used all over the website.)

File: src/docs/resources/platform-adaptations.md [hunk 8]
```diff
@@ -0,0 +1,218 @@
+---
+title: Platform Specific Behaviors and Adaptations
+---
+
+## Adaptation Philosophy
+
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Adaptation philosophy

File: src/docs/resources/platform-adaptations.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+This article mainly covers the automatic adaptations provided by Flutter
+in case 1 on Android and iOS.
+
+For case 2, Flutter bundles the means to produce the appropriate effects of
+the platform conventions but does not adapt automatically when app design
+choices are needed. For a discussion, see [#8410](https://github.com/flutter/flutter/issues/8410#issuecomment-468034023).
+
+Flutter should transparently work for scenario 3. 
+
+## Page Navigation
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Page navigation

Also, please add a sentence before the next-level header.

File: src/docs/resources/platform-adaptations.md [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+
+### Navigation Transitions
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Navigation transitions

File: src/docs/resources/platform-adaptations.md [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+        <img src="../../images/platform-adaptations/navigation-ios-modal.gif" class="figure-img img-fluid rounded" alt="An animation of the bottom-up style present page transition on iOS" />
+        <figcaption class="figure-caption">
+          iOS present transition
+        </figcaption>
+      </figure>
+    </div>
+  </div>
+</div>
+
+### Platform-Specific Transition Details
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Platform-specific transition details

File: src/docs/resources/platform-adaptations.md [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+        <img src="../../images/platform-adaptations/navigation-ios-nav-bar.gif" class="figure-img img-fluid rounded" alt="An animation of the nav bar transitions during a page transition on iOS" />
+        <figcaption class="figure-caption">
+          iOS Nav Bar
+        </figcaption>
+      </figure>
+    </div>
+  </div>
+</div>
+
+### Back Navigation
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Back navigation

File: src/docs/resources/platform-adaptations.md [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+
+On **Android**, the OS back button, by default, is sent to Flutter and pops the top
+route of the [WidgetsApp](https://docs.flutter.io/flutter/widgets/WidgetsApp-class.html)'s
+Navigator.
+
+On **iOS**, an edge swipe gesture can be used to pop the top route.
+
+## Scrolling
+
+### Physics Simulation
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Physics simulation

Can you add a sentence here, before the L3 header? Because a) there should be one and b) anchors don't work right if there's no text.  (This is a global change. I've gone back and looked for it.)

File: src/docs/resources/platform-adaptations.md [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+
+### Overscroll Behavior
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Overscroll behavior

File: src/docs/resources/platform-adaptations.md [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+used.
+
+## Iconography
+
+When using the Material package, certain icons automatically show different
+graphics depending on the platform. For instance, the overflow button's 3 dots
+are vertical on iOS and horizontal on Android. The back button is a simple
+chevron on iOS and has a stem/shaft on Android.
+
+## Haptic Feedback
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Haptic feedback

I know what haptic feedback is, because I recently learned it when my daughter was going through college, but will most ppl?

File: src/docs/resources/platform-adaptations.md [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+The Material and Cupertino packages automatically trigger the platform
+appropriate haptic feedback in certain scenarios.
+
+For instance, a word selection via text field long-press triggers a 'buzz'
+vibrate on Android and not on iOS.
+
+Scrolling through picker items on iOS triggers a 'light impact' knock and
+no feedback on Android.
+
+## Text Editing
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Text editing

And add a sentence here, too.

File: src/docs/resources/platform-adaptations.md [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+
+### Keyboard Gesture Navigation
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Keyboard gesture navigation

File: src/docs/resources/platform-adaptations.md [hunk 18]
```diff
@@ -0,0 +1,218 @@
...
+
+On **Android**, horizontal swipes can be made on the soft keyboard's spacebar
+to move the cursor in Material and Cupertino text fields.
+
+On **iOS** devices with 3D Touch capabilities, a force-press-drag gesture
+could be made on the soft keyboard to move the cursor in 2D via a floating
+cursor. This works on both Material and Cupertino text fields.
+
+### Text Selection Toolbar
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Text selection toolbar

File: src/docs/resources/platform-adaptations.md [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+
+With **Material on Android**, the Android style selection toolbar is shown when
+a text selection is made in a text field.
+
+With **Material on iOS** or when using **Cupertino**, the iOS style selection
+toolbar is shown whe a text selection is made in a text field.
+
+### Single Tap Gesture
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Single tap gesture

File: src/docs/resources/platform-adaptations.md [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+
+With **Material on Android**, a single tap in a text field puts the cursor at
+the location of the tap.
+
+With **Material on iOS** or when using **Cupertino**, a single tap in a text
+field puts the cursor at the nearest edge of the word tapped. 
+
+### Long-Press Gesture
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Long-press gesture

File: src/docs/resources/platform-adaptations.md [hunk 21]
```diff
@@ -0,0 +1,218 @@
...
+
+With **Material on Android**, a long press selects the word under the long
+press. The selection toolbar is shown upon release.
+
+With **Material on iOS** or when using **Cupertino**, a long press places the
+cursor at the location of the long pres. The selection toolbar is shown upon
+release.
+
+### Long-Press-Drag Gesture
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Long-press drag gesture   (I'd eliminate that second hyphen.)

File: src/docs/resources/platform-adaptations.md [hunk 22]
```diff
@@ -0,0 +1,218 @@
...
+
+With **Material on Android**, dragging while holding the long press expands
+the words selected.
+
+With **Material on iOS** or when using **Cupertino**, dragging while holding
+the long press moves the cursor.
+
+### Double Tap Gesture
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: => Double tap gesture

File: src/docs/resources/platform-adaptations.md [hunk 23]
```diff
@@ -0,0 +1,218 @@
...
+
+### Back Navigation
+
+On **Android**, the OS back button, by default, is sent to Flutter and pops the top
+route of the [WidgetsApp](https://docs.flutter.io/flutter/widgets/WidgetsApp-class.html)'s
+Navigator.
+
+On **iOS**, an edge swipe gesture can be used to pop the top route.
+
+## Scrolling
```
Comment (sfshaza2) [inline src/docs/resources/platform-adaptations.md]: Please add a sentence before the level 3 header.

**Comment by xster** [comment]
Thanks for the quick review!

File: src/_data/sidenav.yml (hunk 1 above)
Comment (xster) [reply src/_data/sidenav.yml]: Done

File: src/docs/resources/platform-adaptations.md (hunk 2 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: Changed here and everywhere below to capitalize first word only.

File: src/docs/resources/platform-adaptations.md (hunk 3 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: Ah thanks!

File: src/docs/resources/platform-adaptations.md (hunk 4 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: This part is actually kinda in the gray area. I just removed 3

File: src/docs/resources/platform-adaptations.md (hunk 5 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: Done

File: src/docs/resources/platform-adaptations.md (hunk 6 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: done

File: src/docs/resources/platform-adaptations.md (hunk 7 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: thanks

File: src/docs/resources/platform-adaptations.md (hunk 9 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: Added paragraph everywhere between H2 and H3

File: src/docs/resources/platform-adaptations.md (hunk 15 above)
Comment (xster) [reply src/docs/resources/platform-adaptations.md]: I think so. It's been in the first smartphones.

**Comment by xster** [comment]
Thanks for going through the text. Do you have any feedback on the format of the videos? Otherwise, I'll start capturing the rest of the clips in gifs too.

**Comment by xster** [comment]
ok, all the assets are filled in

also cc @willlarche and @HansMuller FYI

**Comment by willlarche** [comment]
This is so great!!!!

Are you sure iOS is really doing SF font there? I can't tell a different. But I've always had that problem with Roboto and SF.

**Comment by xster** [comment]
I can't see the difference either 😂🤦🏻‍♀️
Though in debug, the render objects says that's how it's painting them so I'll trust it :)

**Comment by willlarche** [comment]
I think the fonts are all good. Looks like there's slightly more air around each glyph in Roboto
